File: src/dataset/cpp/queen-attack_exercism-felbit/claude/ClaudeSonnet4_queen_attack_v1.cpp
#include "queen_attack.h"

namespace queen_attack
{
    chess_board::chess_board() : m_white(0, 3), m_black(7, 3) {}
    
    chess_board::chess_board(const std::pair<int, int> &white, const std::pair<int, int> &black) : m_white(white), m_black(black)
    {
        if (m_white == m_black) 
            throw std::domain_error("The two queens cannot stand on the same position.");
    }

    std::pair<int, int> chess_board::black(void) const
    {
        return m_black;
    }

    std::pair<int, int> chess_board::white(void) const
    {
        return m_white;
    }

    bool chess_board::can_attack() const
    {
        const int row_diff = m_white.first - m_black.first;
        const int col_diff = m_white.second - m_black.second;
        
        return (row_diff == 0) || (col_diff == 0) || (row_diff == col_diff) || (row_diff == -col_diff);
    }
}

File: src/dataset/typescript/queen-attack_exercism-samajammin/claude/ClaudeSonnet4_queen-attack_v1.ts
const BOARD_SIZE = 8;
type Coordinates = [number, number];

export default class QueenAttack {
  white: Coordinates;
  black: Coordinates;

  constructor({ white, black }: { white: Coordinates; black: Coordinates }) {
    if (white[0] === black[0] && white[1] === black[1]) {
      throw 'Queens cannot share the same space';
    }
    this.white = white;
    this.black = black;
  }

  toString() {
    const result: string[] = [];
    
    for (let row = 0; row < BOARD_SIZE; row++) {
      const rowStr: string[] = [];
      for (let col = 0; col < BOARD_SIZE; col++) {
        if (row === this.white[0] && col === this.white[1]) {
          rowStr.push('W');
        } else if (row === this.black[0] && col === this.black[1]) {
          rowStr.push('B');
        } else {
          rowStr.push('_');
        }
      }
      result.push(rowStr.join(' '));
    }
    
    return result.join('\n') + '\n';
  }

  canAttack() {
    const rowDiff = Math.abs(this.white[0] - this.black[0]);
    const colDiff = Math.abs(this.white[1] - this.black[1]);

    return rowDiff === 0 || colDiff === 0 || rowDiff === colDiff;
  }
}